netket/logging.py

```diff
@@ -80,7 +85,8 @@ def __call__(self, step, item, machine):
 
     def _flush_log(self):
         with open(self._prefix + ".log", "w") as outfile:
-            _json.dump(self._json_out, outfile)
+            log_data = _tree_map(_to_json, self._json_out)
```

Comment:
In my opinion this line should go to the `__call__` function, right before `        self._json_out["Output"].append(item)`, so that it is oney called once per logged message.

`flush` should really only flush `_json_out` to a file.